### PR TITLE
Recording time fragment

### PR DESCRIPTION
**Pull request by TimGels**
I have also fixed the binding in the settings activity.

### CODE REVIEW

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt [hunk 1]
```diff
@@ -15,50 +16,53 @@ import kotlinx.coroutines.launch
...
-    private var binding: ActivitySettingsBinding? = null
+    private lateinit var binding: ActivitySettingsBinding
 
     private var keyStore = KeyStoreHelper("connectToServer")
 
-    lateinit var connectionLiveData: ConnectionLiveData
+    private var isFragmentDurationValid: Boolean = false
 
-    lateinit var connectBtn : Button
+    lateinit var connectionLiveData: ConnectionLiveData
```
Comment (JohannesKauffmann) [inline app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:25]: Can you change these around? That way, the connectionLiveData won't be seen as a change. This in order to minimize merge conflicts later.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt [hunk 2]
```diff
@@ -15,50 +16,53 @@ import kotlinx.coroutines.launch
...
 
     private lateinit var sharedPreferences: SharedPreferences
     var pwdIV: String? = null
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_settings)
+        sharedPreferences =
+            this.getSharedPreferences("com.camerasecuritysystem.client", Context.MODE_PRIVATE)
```
Comment (JohannesKauffmann) [inline app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:33]: Same with this one.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt [hunk 3]
```diff
@@ -15,50 +16,53 @@ import kotlinx.coroutines.launch
...
+                    if (!connection.isConnected()) {
                         connection.connectIfPossible(context)
 
-                        Log.e("Connection", "Connecting is possible: ${connection.connectIfPossible(context)}")
-                    }else{
+                        Log.e(
+                            "Connection",
+                            "Connecting is possible: ${connection.connectIfPossible(context)}"
+                        )
+                    } else {
```
Comment (JohannesKauffmann) [inline app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:65]: This is also something that will very likely give merge conflicts with my branch and has no functional change here, so if you could revert this, great.

File: app/src/main/res/values/preferences.xml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="utf-8"?>
+<resources>
+    <integer name="default_fragment_recording_seconds">5</integer>
+    <string name="fragment_recording_seconds">fragment_recording_seconds</string>
```
Comment (JohannesKauffmann) [inline app/src/main/res/values/preferences.xml:4]: why not in strings.xml? The other preference id's are also there.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt [hunk 5]
```diff
@@ -125,8 +214,10 @@ class SettingsActivity : AppCompatActivity(),
 
         sharedPreferences.edit().putString(resources.getString(R.string.port), port).apply()
 
-        sharedPreferences.edit().putString(resources.getString(R.string.ip_address), ipAddress).apply()
+        sharedPreferences.edit().putString(resources.getString(R.string.ip_address), ipAddress)
+            .apply()
 
-        sharedPreferences.edit().putString(resources.getString(R.string.camera_id), cameraID).apply()
+        sharedPreferences.edit().putString(resources.getString(R.string.camera_id), cameraID)
+            .apply()
```
Comment (JohannesKauffmann) [inline app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:221]: No functional change...?

**Review by JohannesKauffmann — CHANGES_REQUESTED** [review]

File: app/src/main/res/values/preferences.xml (hunk 4 above)
Comment (TimGels) [reply app/src/main/res/values/preferences.xml:4]: I don't like to put preferences that I use in a strings XML file that I use for strings

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 2 above)
Comment (TimGels) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:33]: No I will keep the shared preferences there, I like them to be initialized before the content gets set on the view.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 1 above)
Comment (TimGels) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:25]: Done

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 3 above)
Comment (TimGels) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:65]: Altho I will proceed to revert this, the formatter changes this to the way I committed it anyway. So I don't see much issues to be honest.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 5 above)
Comment (TimGels) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:221]: It was too long for the formatter, it put them on a newline. I will revert this, but the formatter its something we want to use before committing.

Comment (TimGels) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt]: Done

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 3 above)
Comment (TimGels) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt]: Done

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 2 above)
Comment (JohannesKauffmann) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:33]: There is another reason for moving it down: In all the other activities, the view and bingings gets set or initialized first, and then the rest of the business logic. I would rather keep that consistent with the rest of the code.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt (hunk 3 above)
Comment (JohannesKauffmann) [reply app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt]: Then the issue is that a) we don't have a formatter or predefined formatting style under VC and b) we don't use the formatter

File: app/src/main/res/values/preferences.xml (hunk 4 above)
Comment (TimGels) [reply app/src/main/res/values/preferences.xml:4]: I can do that in a next "chore" pr.

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt [hunk 6]
```diff
@@ -15,50 +16,50 @@ import kotlinx.coroutines.launch
...
 
+        binding.connectBtn.setOnClickListener {
             val context = this.applicationContext
 
             //Try to setup a connection
             GlobalScope.launch {
                 try {
                     val connection = ServerConnection.getInstance()
-                    if (!connection.isConnected()){
+                    if (!connection.isConnected()) {
```
Comment (JohannesKauffmann) [inline app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:58]:
```suggestion
                    if (!connection.isConnected()){
```

File: app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt [hunk 7]
```diff
@@ -15,50 +16,50 @@ import kotlinx.coroutines.launch
...
                         connection.connectIfPossible(context)
 
-                        Log.e("Connection", "Connecting is possible: ${connection.connectIfPossible(context)}")
-                    }else{
+                        Log.e("Connection","Connecting is possible: ${connection.connectIfPossible(context)}")
+                    } else {
```
Comment (JohannesKauffmann) [inline app/src/main/kotlin/com/camerasecuritysystem/client/SettingsActivity.kt:62]:
```suggestion
                        Log.e("Connection", "Connecting is possible: ${connection.connectIfPossible(context)}")
                    }else{
```

**Review by JohannesKauffmann — APPROVED** [review]

**Pull request merged**